core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -575,27 +575,35 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
   private def changeToWitnessUTXO(
       transactionOutput: TransactionOutput): InputPSBTMap = {
     val newElements = transactionOutput.scriptPubKey match {
-      case _: WitnessScriptPubKey =>
+      case rs: WitnessScriptPubKey
+          if !rs.isInstanceOf[BIP143VulnerableScriptPubKey] =>
         filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
       case _: P2SHScriptPubKey =>
-        if (redeemScriptOpt.isDefined && redeemScriptOpt.get.redeemScript
-              .isInstanceOf[WitnessScriptPubKey]) {
-          filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
+        if (redeemScriptOpt.isDefined) {
+          val redeemScript = redeemScriptOpt.get.redeemScript
+          if (redeemScript
+                .isInstanceOf[WitnessScriptPubKey] && !redeemScript
+                .isInstanceOf[BIP143VulnerableScriptPubKey]) {
+            filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
+          } else {
+            elements
+          }
         } else {
           elements
         }
-      case _: P2PKHScriptPubKey | _: P2PKScriptPubKey |
-          _: P2PKWithTimeoutScriptPubKey | _: MultiSignatureScriptPubKey |
-          EmptyScriptPubKey | _: LockTimeScriptPubKey |
-          _: NonStandardScriptPubKey | _: WitnessCommitment |
-          _: ConditionalScriptPubKey =>
+      case _: BIP143VulnerableScriptPubKey | _: P2PKHScriptPubKey |
```

Comment:
I'm not convinced this new case is needed (should still be exhaustive without), also refer to my not liking this trait above)